project/src/main/ui/level-select/hookable-grade-label.gd

```diff
@@ -0,0 +1,120 @@
+class_name HookableGradeLabel
+extends Node2D
+"""
+Shows a grade label for a specific level.
+
+This can be something like 'S' or 'B+' if a level has been cleared, or something like a lock/key icon for levels
+which haven't yet been cleared.
+"""
+
+# the level button this grade label applies to
+var button: LevelSelectButton setget set_button
+
+var _crown_texture: Texture = preload("res://assets/main/ui/level-select/crown.png")
+var _key_texture: Texture = preload("res://assets/main/ui/level-select/key.png")
+var _locked_texture: Texture = preload("res://assets/main/ui/level-select/locked.png")
+
+"""
+Assigns this label's button, and updates the label's appearance.
+"""
+func set_button(new_button: LevelSelectButton) -> void:
+	button = new_button
+	
+	button.get_node("GradeHook").remote_path = button.get_node("GradeHook").get_path_to(self)
+	if not button.is_connected("lowlight_changed", self, "_on_LevelSelectButton_lowlight_changed"):
+		button.connect("lowlight_changed", self, "_on_LevelSelectButton_lowlight_changed")
+	if not button.is_connected("tree_exited", self, "_on_LevelSelectButton_tree_exited"):
+		button.connect("tree_exited", self, "_on_LevelSelectButton_tree_exited")
```

Comment:
While it will never come up with how our code is written, we should handle the case where the button is already assigned.

We could throw an error, or do the legwork of disconnecting the listeners/remote_path.

Reply:
The setter now handles the case where the button is already assigned (and, the case where the button is being redundantly set, which makes our code a little cleaner.)